test/BlazorSignalR.Test.Client/Pages/ChatComponent.cs

```diff
@@ -13,52 +14,76 @@ namespace BlazorSignalR.Test.Client.Pages
 {
     public class ChatComponent : ComponentBase
     {
-        [Inject] private HttpClient _http { get; set; }
-        [Inject] private ILogger<ChatComponent> _logger { get; set; }
-        [Inject] private IJSRuntime _jsRuntime { get; set; }
-        internal string _toEverybody { get; set; }
-        internal string _toConnection { get; set; }
-        internal string _connectionId { get; set; }
-        internal string _toMe { get; set; }
-        internal string _toGroup { get; set; }
-        internal string _groupName { get; set; }
-        internal List<string> _messages { get; set; } = new List<string>();
+        [Inject] private IJwtTokenResolver JwtTokenResolver { get; set; }
+        [Inject] private ILogger<ChatComponent> Logger { get; set; }
+        [Inject] private IJSRuntime JsRuntime { get; set; }
+        [Inject] private IUriHelper UriHelper { get; set; }
+
+        // Needed to detect prerendering
+        [Inject] private IComponentContext ComponentContext { get; set; }
+
+        internal string ToEverybody { get; set; }
+        internal string ToConnection { get; set; }
+        internal string ConnectionId { get; set; }
+        internal string ToMe { get; set; }
+        internal string ToGroup { get; set; }
+        internal string GroupName { get; set; }
+        internal List<string> Messages { get; set; } = new List<string>();
 
         private IDisposable _objectHandle;
         private IDisposable _listHandle;
         private HubConnection _connection;
 
-        protected override async Task OnInitAsync()
+        private bool _connectionInitalized = false;
+
+        protected override async Task OnAfterRenderAsync()
+        {
+            if (!ComponentContext.IsConnected)
```

Comment:
I'd suggest documenting what this is doing with a comment. People may confuse this connected state.

Reply:
You mean `ComponentContext.IsConnected`? This is because of the pre-rendering that is done if running server-side. I will add some comments to further explain this.

Reply:
Yeah. Just a one line comment might be useful incase someone mistakes it with the blazor signalr connection or something strange.